aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,623 @@
+            } else {
+                stats.total_no = stats.total_no + 1;
+                bucket_table::add(&mut stats.no_votes, token_id, voter_addr);
+            };
+            i = i + 1;
+        };
+    }
+
+    /// Entry function that can be called by anyone
+    public entry fun resolve(proposal_id: u64, nft_dao: address) acquires Proposals, DAO, ProposalVotingStatistics, RegisteredDAOs {
+        // validate if proposal is ready to resolve
+        let dao = borrow_global_mut<DAO>(nft_dao);
+
+        // assert the proposal voting ended
+        let proposals = borrow_global_mut<Proposals>(nft_dao);
+        let proposal = table::borrow_mut(&mut proposals.proposals, proposal_id);
+        let now = timestamp::now_seconds();
+        assert!(now >= proposal.start_time_sec + dao.voting_duration, error::invalid_argument(EPROPOSAL_ENDED));
+
+        // assert the proposal is unresolved yet
+        assert!(proposal.resolution == PROPOSAL_UNRESOLVED, error::invalid_argument(EPROPOSAL_RESOLVED));
+
+        let proposal_stat = &mut borrow_global_mut<ProposalVotingStatistics>(nft_dao).proposals;
+        let voting_stat = table::borrow_mut(proposal_stat, proposal_id);
+        // validate resolve threshold and result
+        let voted = voting_stat.total_no + voting_stat.total_yes;
+        if (voted < dao.resolve_threshold) {
+            // not sufficient token voted
+            proposal.resolution = PROPOSAL_RESOLVED_NOT_PASSED;
+            return
+        };
+        let passed = if (voting_stat.total_yes > voting_stat.total_no) {true} else {false};
+        if (passed) {
+            let accts = &borrow_global<RegisteredDAOs>(@dao_platform).res_acct_to_src_acct;
+            assert!(table::contains(accts, nft_dao), error::not_found(ERESOURCE_ACCT_NOT_EXIST));
+            let src_acct = *table::borrow(accts, nft_dao);
+
+            let function_name = proposal.function_name;
+            if (function_name == string::utf8(b"transfer_fund")) {
+                // This is very dangerous.
+                // We should exploring have server side dynamic compiling and deploying DAO contracts in each DAO's own accounts.
+                let res_signer = get_dao_signer(src_acct);
+                let dst_addr = property_map::read_address(&proposal.function_args, &string::utf8(b"dst"));
+                let amount = property_map::read_u64(&proposal.function_args, &string::utf8(b"amount"));
+                transfer_fund(&res_signer, dst_addr, amount);
+            } else {
+               assert!(function_name == string::utf8(b"no_op"), error::invalid_argument(ENOT_SUPPROTED_FUNCTION));
+            };
+            proposal.resolution = PROPOSAL_RESOLVED_PASSED;
+        } else {
+            proposal.resolution = PROPOSAL_RESOLVED_NOT_PASSED;
+        };
+    }
+
+    /// DAO creator can delegate its DAO to another admin account
+    public entry fun delegate_dao(admin: &signer, new_admin: address)acquires RegisteredDAOs {
```

Comment:
Two points here:
1. I think we should call the account admin everywhere instead of DAO creator, which is not very clear regarding the role of the account.
2. This would also come up in security audits, but admin transferring should always be done in two step (1. set a pending admin and 2. the pending admin accepts, which officially changes the admin). This is for preventing accidental setting of the admin to the wrong address, which happens quite a lot in crypto

Reply:
For DAO transfer, let's do it in a follow up PR. I guess the current PR is already large enough?